Sources/Datadog/RUM/RUMMonitor/Scopes/RUMResourceScope.swift

```diff
@@ -128,6 +128,13 @@ internal class RUMResourceScope: RUMScope {
         }
         let resourceType: RUMResourceType = resourceKindBasedOnRequest ?? command.kind
 
+        let ciTest: RUMResourceEvent.CiTest? = {
+            if let testID = CITestIntegration.ciTestExecutionID {
+                return RUMResourceEvent.CiTest(testExecutionId: testID)
+            }
+            return nil
+        }()
```

Comment:
We can improve on this repetitive snippet in this PR if we make `CiTest` model "shared" in `RUMDataModels` definition. This way, we will have single model for all events, instead of separate nested model in each event.

Once we have a single model, we can create its single value and pass it through `dependencies` container available in all RUM scopes:
```swift
ciTest: dependencies.ciTest,
```

To make `CiTest` model shared, we need to configure it in `tools/rum-models-generator` and call `make make rum-models-generate`. I did it [here](https://github.com/DataDog/dd-sdk-ios/commit/ad51785d084c8ecc4ff70fd09c20d4efcbf615e6#diff-dd3caa4e6db960eecd95024f6d7df904948b8f53b1026c00dd19a9627317ac3e) - we might consider just merging it.

Reply:
Yes please, merge it. That was the part of the code that looked worse for me, but I didn't know how to improve it.

Reply:
@nachoBonafonte feel free to merge my branch to yours 🙌. I don't want to mess up in your workflow 🙂.

Reply:
Merged your branch and it looks much better now